src/Symfony/Component/DependencyInjection/Compiler/MergeExtensionConfigurationPass.php

```diff
@@ -117,6 +118,15 @@ public function __construct(parent $parameterBag)
         $this->mergeEnvPlaceholders($parameterBag);
     }
 
+    public function resolveValue(mixed $value, array $resolving = []): mixed
+    {
+        if ($value instanceof ParamConfigurator) {
+            $value = (string) $value;
+        }
+
+        return parent::resolveValue($value, $resolving);
```

Comment:
it would be move consistent to me to cast Stringables to string in ParameterBag

Reply:
I didn't plan to fully support Stringables in ParameterBag, what is your idea? Supporting all Stringables as values of ParameterBag and cast them in set()?

Reply:
The reasoning that makes me +0 for this PR is `Stringable` should be handled transparently when `string` is allowed.
I'm not sure I agree with the initial motivation for this PR: using config builders outside of the PHP-DSL.
Dealing only with `ParamConfigurator` here looks like an abstraction leak to me.

Reply:
I prefer to close this PR because it derivates too much from my initial motivation which I don't find that great anymore after the different reviews :-)

About supporting Stringable transparently, I think we can't really ignore this comment from @derrabus:
>  A class might be stringable for debug purposes or other reasons.